core/model/src/main/java/org/eclipse/rdf4j/model/util/RDFCollections.java

```diff
@@ -208,7 +208,7 @@ public static void consumeCollection(Iterable<?> values, Resource head, Consumer
 
 		Resource current = head != null ? head : vf.createBNode();
 
-		Statements.consume(vf, current, RDF.TYPE, RDF.LIST, consumer, contexts);
+		//Statements.consume(vf, current, RDF.TYPE, RDF.LIST, consumer, contexts);
```

Comment:
IMO this hides the issue rather than fixing it. I don't have a problem with taking this line out (it's not _required_ to have this type statement for a collection), but I think we want a more comprehensive fix, as well.

I think what you want to look at is the `TurtleWriter` class. The handling of writing lists is in the `handleStatementInternal` method (line 299-321). In particular, take a look at line 307 - this is where it decides if it will process the statement as part of a list:

```java
				if ((pred.equals(RDF.FIRST) || pred.equals(RDF.REST)) && isWellFormedCollection(subj)) {
					// we only use list shorthand syntax if the collection is considered well-formed
					handleList(st, canShortenObjectBNode);
```

My guess is the if-condition is overlooking the fact that it can also be a `rdf:type rdf:List` statement that should be handled as a list. As for _how_ it should handle it: my gut feel is that it should simply skip this statement.